### PR TITLE
refine balance to replica granuity

### PR DESCRIPTION
**Pull request by MrPresent-Han**
related: #23454

### CODE REVIEW

**Comment by MrPresent-Han** [comment]
rerun ut

**Comment by MrPresent-Han** [comment]
rerun ut

File: internal/querycoordv2/task/scheduler.go [hunk 1]
```diff
@@ -782,6 +781,13 @@ func (scheduler *taskScheduler) checkStale(task Task) bool {
 			log.Warn("the task is stale, the target node is offline")
 			return true
 		}
+		if action.Type() == ActionTypeGrow {
+			isStoppingNode, _ := scheduler.nodeMgr.IsStoppingNode(action.Node())
+			if isStoppingNode {
+				log.Warn("task stale due to dst node for growing has been stopping", zap.Int64("dstNode", action.Node()))
```
Comment (yah01) [inline internal/querycoordv2/task/scheduler.go:787]: the "nodeID" has been added into the log at line 777

**Review by yah01 — COMMENTED** [review]
/lgtm except the log part

File: internal/querycoordv2/task/scheduler.go (hunk 1 above)
Comment (congqixia) [reply internal/querycoordv2/task/scheduler.go:787]: NIT comment
Please fix it later

**Review by congqixia — APPROVED** [review]
/kind improvement
/lgtm

**Pull request merged**